### PR TITLE
Enable bamboo HAProxy autoconfiguration and discovery.

### PR DESCRIPTION
**Pull request by johngian**
- Install bamboo dependencies
- Install haproxy
- Add bamboo config
- Add haproxy template

### CODE REVIEW

File: roles/master/files/go-path.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+export GOPATH="/usr/share/go/"
```
Comment (akatsoulas) [inline roles/master/files/go-path.sh]: Maybe you should use the environment module for this.

File: roles/master/files/haproxy_template.cfg [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        errorfile 400 /etc/haproxy/errors/400.http
+        errorfile 403 /etc/haproxy/errors/403.http
+        errorfile 408 /etc/haproxy/errors/408.http
+        errorfile 500 /etc/haproxy/errors/500.http
+        errorfile 502 /etc/haproxy/errors/502.http
+        errorfile 503 /etc/haproxy/errors/503.http
+        errorfile 504 /etc/haproxy/errors/504.http
+
+
+# Template Customization
```
Comment (akatsoulas) [inline roles/master/files/haproxy_template.cfg]: Add a comment that it's not a `jinja` template.

File: roles/master/tasks/main.yml [hunk 3]
```diff
@@ -95,3 +95,64 @@
     mode=0644
   notify:
     - restart marathon
+
+- name: Make sure haproxy /var/run folder exists
```
Comment (akatsoulas) [inline roles/master/tasks/main.yml]: Isn't this /var/run/haproxy?

File: roles/master/tasks/main.yml [hunk 4]
```diff
@@ -95,3 +95,64 @@
...
+- name: Set GOPATH for all users
+  copy:
+    src=go-path.sh
+    dest=/etc/profile.d/
+    mode=0755
+
+- name: Install fpm using ruby gems
+  command: gem install fpm
+
+- name: Fetch bamboo latest release from github
```
Comment (akatsoulas) [inline roles/master/tasks/main.yml]: s/latest/with_version

File: roles/master/handlers/main.yml [hunk 5]
```diff
@@ -17,3 +17,9 @@
     name: marathon
     enabled: true
     state: restarted
+
+- name: restart bamboo
+  service:
+    name: bamboo
```
Comment (johngian) [inline roles/master/handlers/main.yml]: Service is `bamboo-server`.

File: roles/master/tasks/main.yml [hunk 6]
```diff
@@ -95,3 +95,64 @@
...
+    state: directory
+
+- name: Install bamboo requirements
+  apt: name={{ item }} state=present
+  with_items:
+    - git
+    - ruby-all-dev
+    - build-essential
+    - golang
+    - haproxy
```
Comment (akatsoulas) [inline roles/master/tasks/main.yml]: Maybe use another yml file for this and include it in here?

File: roles/master/tasks/main.yml [hunk 7]
```diff
@@ -95,3 +95,64 @@
...
+
+- name: Set GOPATH for all users
```
Comment (akatsoulas) [inline roles/master/tasks/main.yml:114]: Same as above

File: roles/master/tasks/main.yml [hunk 8]
```diff
@@ -95,3 +95,64 @@
...
+    mode=0755
+
+- name: Install fpm using ruby gems
+  command: gem install fpm
+
+- name: Fetch bamboo latest release from github
+  git:
+    repo=https://github.com/QubitProducts/bamboo.git
+    dest=/root/bamboo
+    version=v0.2.12
```
Comment (akatsoulas) [inline roles/master/tasks/main.yml]: Maybe using a var here is a better idea.

File: roles/master/tasks/main.yml [hunk 9]
```diff
@@ -95,3 +95,64 @@
...
+
+- name: Build bamboo from source
+  environment:
+    GOPATH: /usr/share/go/
+  shell: go build bamboo.go
+  args:
+    chdir: /root/bamboo
```
Comment (akatsoulas) [inline roles/master/tasks/main.yml]: what about s/root/tmp?

**Comment by akatsoulas** [comment]
code looks good, we can optimize a few things later. r+wc so that we are no longer blocked from this

**Pull request merged**